azure-communication-ui/calling/src/main/java/com/azure/android/communication/ui/calling/redux/reducer/CallDiagnosticsReducer.kt

```diff
@@ -0,0 +1,31 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.android.communication.ui.calling.redux.reducer
+
+import com.azure.android.communication.ui.calling.redux.action.Action
+import com.azure.android.communication.ui.calling.redux.action.CallDiagnosticsAction
+import com.azure.android.communication.ui.calling.redux.state.CallDiagnosticsState
+
+internal interface CallDiagnosticsReducer : Reducer<CallDiagnosticsState>
+
+internal class CallDiagnosticsReducerImpl : CallDiagnosticsReducer {
+
+    override fun reduce(
+        state: CallDiagnosticsState,
+        action: Action
+    ): CallDiagnosticsState {
+        return when (action) {
+            is CallDiagnosticsAction.NetworkQualityCallDiagnosticsUpdated -> {
+                state.copy(networkQualityCallDiagnostic = action.networkQualityUpdated)
+            }
+            is CallDiagnosticsAction.NetworkCallDiagnosticsUpdated -> {
```

Comment:
should we add a action to clear error when user press close? as we do not want to show again when user close?